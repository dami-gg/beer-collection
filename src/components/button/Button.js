import React from 'react';

import './button.scss';

const Button = (props) => (
    <button type={props.type || "button"}
            className={`button button--${props.color} ${props.classes}`}
            onClick={event => props.onClick && props.onClick(event)}>
      {props.children}
    </button>
);

export default Button;
