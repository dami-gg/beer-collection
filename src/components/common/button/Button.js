// @flow

import React, {PureComponent} from 'react';

import './button.scss';

class Button extends PureComponent {
  props: {
    type?: string,
    color: string,
    classes?: string,
    onClick: Function,
    children: any
  };

  render() {
    return (
        <button type={this.props.type || "button"}
                className={`button button--${this.props.color} ${this.props.classes}`}
                onClick={event => this.props.onClick && this.props.onClick(event)}>
          {this.props.children}
        </button>
    );
  }
}

export default Button;
