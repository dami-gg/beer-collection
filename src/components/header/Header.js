// @flow
import type { User } from "../../types/user.types";

import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import Button from "../common/button/Button";

import Logo from "../common/logo/Logo";
import { logout } from "../../helpers/authentication.helpers";

import "./header.scss";

type Props = {
  user: User
};

export class Header extends Component<Props> {
  render() {
    return (
      <div className="header">
        <Link to="/" className="logo-container">
          <Logo collapsed={true} />
        </Link>
        {this.props.user && (
          <div className="logout">
            <Button color="red" onClick={logout}>
              Logout
            </Button>
          </div>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.authentication.user
});

export default connect(mapStateToProps)(Header);
