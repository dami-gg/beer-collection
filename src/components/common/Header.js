// @flow

import React, {Component} from 'react';
import {connect} from 'react-redux';
import {hashHistory} from 'react-router';
import firebase from 'firebase';

import './header.scss';
import {Button} from 'react-bootstrap';
import logo from '../../assets/images/logo.png';

class Header extends Component {
  navigateToHome():void {
    hashHistory.push('/');
  }

  logout():void {
    firebase.auth().signOut()
        .then(() => hashHistory.push('/'))
        .catch();
  }

  render() {
    return (
        <div className="header">
          <div className="logo-container" onClick={this.navigateToHome}>
            <div className="logo">
              <img src={logo} alt="logo"/>
            </div>
            <span className="title">Beer collection</span>
          </div>
          {
            this.props.user &&

            <div className="logout">
              <Button bsStyle="danger"
                      onClick={this.logout}>
                Logout
              </Button>
            </div>
          }
        </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.navigation.user
});

export default connect(
    mapStateToProps
)(Header);

