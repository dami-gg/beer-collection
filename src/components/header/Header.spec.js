import React from "react";
import { Header } from "./Header";

import { StaticRouter as Router, Link } from "react-router-dom";
import Button from "../common/button/Button";

import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Header", () => {
  let component, element, mock;

  beforeEach(mocks);

  it("should render without crashing", () => {
    shallow(<Header />);
  });

  it("should match the snapshot", () => {
    component = renderer.create(
      <Router context={mock.context}>
        <Header />
      </Router>
    );
    let json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  describe("render", () => {
    it("should render a Link component", () => {
      render();

      element = component.find(Link);

      expect(element.length).toEqual(1);
    });

    it("should render a logout Button component when there is a user", () => {
      render();

      element = component.find(Button);

      expect(element.length).toEqual(1);
    });

    it("should not render a logout Button component when there is no user", () => {
      delete mock.props.user;
      render();

      element = component.find(Button);

      expect(element.length).toEqual(0);
    });
  });

  function render() {
    component = shallow(<Header user={mock.props.user} />);
  }

  function mocks() {
    mock = {
      context: {},
      props: {
        user: { id: 1, name: "Name" }
      }
    };
  }
});
