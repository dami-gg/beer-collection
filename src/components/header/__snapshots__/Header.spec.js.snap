// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header dumb component should match the snapshot 1`] = `
<div
  className="header"
>
  <a
    className="logo-container"
    href="/"
    onClick={[Function]}
  >
    <div
      className="logo logo--collapsed"
    >
      <div
        className="logo__image-wrapper"
      >
        <img
          alt="logo"
          className="logo__image"
          src="logo.png"
        />
      </div>
      <h1
        className="logo__title"
        data-text="Beer collection"
      >
        Beer collection
      </h1>
    </div>
  </a>
  <div
    className="notification-container notification-container-empty"
  >
    <span />
  </div>
</div>
`;
